godot_project/editor/rendering/atomic_structure_renderer/representation/springs_representation/springs_representation.gd

```diff
@@ -203,13 +205,18 @@ func lowlight_bonds(_in_bonds_ids: PackedInt32Array) -> void:
 
 
 func highlight_springs(in_springs_to_highlight: PackedInt32Array) -> void:
+	var structure_context: StructureContext = _workspace_context.get_structure_context(_structure_id)
+	var nano_struct: NanoStructure = structure_context.nano_structure
 	for spring_id: int in in_springs_to_highlight:
-		_spring_renderer.change_spring_color(spring_id, COLOR_HIGHLIGHT)
+		var is_selected: bool = structure_context.is_spring_selected(spring_id)
+		_spring_renderer.change_spring_color(spring_id, COLOR_HIGHLIGHT, is_selected)
```

Comment:
Cannot assume is_selected is always `true`?

Reply:
well, we probably can indeed